### PR TITLE
UICAL-206 test AllCalendarView

### PR DESCRIPTION
**Pull request by zburke**
Test `./src/views/AllCalendarView`.

* Elaborate stripes-core mock copied mostly verbatim from ui-users, minus a few things TS barfed on.
* `withHistoryConfiguration` mirrors `withIntlConfiguration` but provides a react-router provider.
* Minor bug fix: localize the "All calendars" pane title

Refs [UICAL-206](https://issues.folio.org/browse/UICAL-206)

### CODE REVIEW

**Comment by zburke** [comment]
Oh dang. I'm guessing the stripes-core mock is breaking other tests. You can hold off on reviewing while I investigate.

File: src/test/__mocks__/stripes-core.mock.ts [hunk 1]
```diff
@@ -1,4 +1,110 @@
...
+      } else if (perm.startsWith('ui-calendar')) {
+        return children;
+      } else if (perm.startsWith('perms')) {
+        return children;
+      } else {
+        return null;
+      }
+    }),
+    Pluggable: jest.fn(({ children }) => [children]),
+    useStripes: () => STRIPES,
```
Comment (ncovercash) [inline src/test/__mocks__/stripes-core.mock.ts:24]:
```suggestion
    useStripes: () => STRIPES,
    useOkapiKy: jest.fn(),
```

File: src/views/AllCalendarView.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ const AllCalendarView: FunctionComponent<Record<string, never>> = () => {
   )?.params?.calendarId;
 
   if (!dataRepository.isLoaded()) {
-    return <LoadingPane paneTitle="All calendars" />;
+    return <LoadingPane paneTitle={intl.formatMessage({ id: 'ui-calendar.allCalendarView.title' })} />;
```
Comment (ncovercash) [inline src/views/AllCalendarView.tsx:42]: Good catch, thanks!

**Review by ncovercash — APPROVED** [review]

File: src/test/__mocks__/stripes-core.mock.ts (hunk 1 above)
Comment (ncovercash) [reply src/test/__mocks__/stripes-core.mock.ts:24]: Oops, didn't realize this was pending as part of a review and wasn't commented earlier; disregard.

**Comment by ncovercash** [comment]
Good to merge as soon as sonar approves! 🙌

**Pull request merged**